shark_turbine/kernel/wave/register_analysis.py

```diff
@@ -9,13 +9,20 @@ def set_register_shape(trace: CapturedTrace, custom: CustomOp) -> None:
     for custom_user in custom.users:
         if isinstance(custom_user, MMA):
             arg_index = custom_user.fx_node.args.index(custom.fx_node)
+            get_thread_shape = lambda index: max(x.size for x in index.values())
             match arg_index:
                 case 0:
-                    custom.fx_node.thread_shape = custom_user.lhs_index[0].size
+                    custom.fx_node.thread_shape = get_thread_shape(
```

Comment:
For the case of lhs_index specifically(we can generalize this to rhs and acc_index), is the logic here since we set MMA_RHS and MMA_ACC to 0s and only set MMA_LHS to 1, only the LHS related dimension will be non zero here?